source/SIL.AppBuilder.Portal.Frontend/src/ui/components/product-artifact/index.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import { compose } from 'recompose';
+import { withData as withOrbit } from 'react-orbitjs';
+
+import ProductIcon from '@ui/components/product-icon';
+import {
+  ProductDefinitionResource,
+  ProductArtifactResource,
+  ProductResource,
+  attributesFor
+} from '@data';
+import Artifact from './artifact';
+import EmptyLabel from '@ui/components/labels/empty';
+import { isEmpty } from '@lib/collection';
+import { withTranslations, i18nProps } from '@lib/i18n';
+
+import Header from './header';
+
+interface IExpectedProps {
+  product: ProductResource;
+}
+
+interface IOwnProps {
+  productDefinition: ProductDefinitionResource;
+  artifacts: ProductArtifactResource[];
+}
+
+interface IState {
+  areArtifactsVisible: boolean;
+}
+
+type IProps =
+  & IOwnProps
+  & i18nProps;
+
+const mapRecordsToProps = (passedProps) => {
```

Comment:
We have been moving this type of functions into the HOC declaration like this

withOrbit(() => {
});

could be a nice refactor